### PR TITLE
Add GET note labels endpoint

### LINKED ISSUE #55: Notes Labels Endpoint
GET /notes/labels

Return all unique note labels in the DB.
Returns a list of all current labels in order to allow user to choose from a list of current labels.

### PR DESCRIPTION
**Pull request by daniel-moon32**
## Summary

Get unique labels for all notes.

## Changes

- Endpoint that returns list of names of labels

## Requests / Responses 

**Request**
(Only for backend)
GET `/notes/labels` Returns list of unique labels

**Response**
(Only for backend)
HTTP/1.1 200 OK

```
{
  "success":true,
  "result":["1V1","Interview"]
}
```

Closes #55

### CODE REVIEW

File: api/src/api/notes.js [hunk 1]
```diff
@@ -45,6 +45,29 @@ router.get(
   }),
 );
 
+//GET /notes/labels
```
Comment (Leundai) [inline api/src/api/notes.js:48]: This is super nitpicky 😅 but putting a space after // helps with consistency like the other comments

**Review by Leundai — APPROVED** [review]
lgtm

File: api/src/api/notes.js [hunk 2]
```diff
@@ -45,6 +45,29 @@ router.get(
...
+router.get(
+  '/labels',
+  requireRegistered,
+  errorWrap(async (req, res) => {
+    const notes = await Note.find({});
+    let labelList = [];
```
Comment (yousefa00) [inline api/src/api/notes.js:54]: This can be `const`

File: api/src/api/notes.js [hunk 3]
```diff
@@ -45,6 +45,29 @@ router.get(
...
+
+    for (const note of notes) {
+      for (const label of note.metaData.labels) {
+        if (!labelList.includes(label)) {
+          labelList.push(label);
+        }
+      }
+    }
+
+    res.status(200).json({
```
Comment (yousefa00) [inline api/src/api/notes.js:64]: Add a `message` response field for the frontend. Not super important but good practice nonetheless. Something like "Notes retrieved successfully."

**Review by yousefa00 — APPROVED** [review]
beautiful 🚀

**Pull request merged**